src/main/java/tutorease/address/logic/commands/DeleteLessonCommand.java

```diff
@@ -45,12 +46,12 @@ public CommandResult execute(Model model) throws CommandException {
 
         int listIndex = targetIndex.getZeroBased();
 
-        if (listIndex >= model.getLessonScheduleSize()) {
+        if (listIndex >= model.getFilteredLessonList().size()) {
```

Comment:
Should we have a getFileredLessonListSize(), currently it might violate LOD

Reply:
Maybe this line `if (listIndex >= model.getFilteredLessonList().size()) {` can be written as 
`if (listIndex >= model.getFileredLessonListSize()) {`
For LOD

Reply:
Yes i agree with what Norbert mentioned here. Based on textbook, we want to refrain from violating LOD. 🤔